File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/admin/users/AdminUsersSetRegularRequest.java
package com.github.seratch.jslack.api.methods.request.admin.users;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

/**
 * https://api.slack.com/methods/admin.users.setRegular
 */
public class AdminUsersSetRegularRequest implements SlackApiRequest {

    /**
     * Authentication token bearing required scopes.
     */
    private String token;

    /**
     * Workspace Id
     */
    private String teamId;

    /**
     * The ID of the user to designate as a regular user.
     */
    private String userId;

    AdminUsersSetRegularRequest(String token, String teamId, String userId) {
        this.token = token;
        this.teamId = teamId;
        this.userId = userId;
    }

    public static AdminUsersSetRegularRequestBuilder builder() {
        return new AdminUsersSetRegularRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public String getTeamId() {
        return this.teamId;
    }

    public String getUserId() {
        return this.userId;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setTeamId(String teamId) {
        this.teamId = teamId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public static class AdminUsersSetRegularRequestBuilder {
        private String token;
        private String teamId;
        private String userId;

        AdminUsersSetRegularRequestBuilder() {
        }

        public AdminUsersSetRegularRequest.AdminUsersSetRegularRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public AdminUsersSetRegularRequest.AdminUsersSetRegularRequestBuilder teamId(String teamId) {
            this.teamId = teamId;
            return this;
        }

        public AdminUsersSetRegularRequest.AdminUsersSetRegularRequestBuilder userId(String userId) {
            this.userId = userId;
            return this;
        }

        public AdminUsersSetRegularRequest build() {
            return new AdminUsersSetRegularRequest(token, teamId, userId);
        }

        public String toString() {
            return "AdminUsersSetRegularRequest.AdminUsersSetRegularRequestBuilder(token=" + this.token + ", teamId=" + this.teamId + ", userId=" + this.userId + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/admin/users/AdminUsersSetRegularRequest.kt
package work.socialhub.kslack.api.methods.request.admin.users

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest

/**
 * https://api.slack.com/methods/admin.users.setRegular
 */
class AdminUsersSetRegularRequest(
    /** Authentication token bearing required scopes. */
    override var token: String?,
    /** Workspace Id */
    var teamId: String?,
    /** The ID of the user to designate as a regular user. */
    var userId: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("team_id", teamId)
            it.addParam("user_id", userId)
        }
    }
}